exec/java-exec/src/main/java/org/apache/drill/exec/store/easy/text/compliant/TextParsingSettings.java

```diff
@@ -57,6 +56,14 @@ public void set(TextFormatConfig config){
     this.delimiter = bSafe(config.getFieldDelimiter(), "fieldDelimiter");
     this.comment = bSafe(config.getComment(), "comment");
     this.skipFirstLine = config.isSkipFirstLine();
+    //TODO: avoid specific extension based check
```

Comment:
You can make this work correctly by using a format plugin config. You can add csvh here:

https://github.com/apache/drill/blob/master/exec/java-exec/src/main/resources/bootstrap-storage-plugins.json

Basically, create a new item in the map. Note how the text formats are a subset of the TextFormatConfig class. You can add a new property that tells you to perform header extraction. See the properties here [1] that correspond to the format properties: [1] https://github.com/apache/drill/blob/master/exec/java-exec/src/main/java/org/apache/drill/exec/store/easy/text/TextFormatPlugin.java#L134